### PR TITLE
NickAkhmetov/Homepage Hero

### PR DESCRIPTION
**Pull request by NickAkhmetov**
This PR

- Implements the new homepage hero
  - On desktop, this is a carousel triggered by hovering or focusing over one of the "tabs"
  - On mobile, the carousel slides collapse into a stack
- Standardizes some of our custom SVGs' dimensions to 24x24
  - This makes styling them more consistent with the MUI icons
- Adds the ability to set icon font size in rems
  - Convenience and fine-tuning odd-sized icons
- Adds `elevated` and `halfRound` chip variants to reflect homepage action styles
- Upgrades to Storybook 8 and resolves issues loading SVGs as React components
  - Fixing this appears to have broken Cypress and Jest...

<details><summary>Desktop preview</summary>
<p>

https://github.com/hubmapconsortium/portal-ui/assets/19957804/5ce144a9-50e5-4557-b5ae-02589893f0ae

</p>
</details> 

<details><summary>Mobile preview</summary>
<p>

image

</p>
</details> 

<details><summary>Storybook for timeline</summary>
<p>

image

</p>
</details> 

Misc TODOs:
- [x] Investigate how to fix storybook imports of svg files
- ~~Fix outbound link appearance for AVR on mobile (outbound link icon is all the way to the right)~~ - made [CAT-708](https://hms-dbmi.atlassian.net/browse/CAT-708) to track this
- [x] `style.js` -> `style.ts` conversion
- [x] Add real images
- [x] Optimize images
- [x] Fix Cypress

[CAT-708]: https://hms-dbmi.atlassian.net/browse/CAT-708?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by NickAkhmetov** [comment]
I fixed the Jest tests and the maintenance webpack conf issues which were causing CI to fail; however, it appears that the test publication we rely on for e2e tests has been removed. I have confirmed e2e tests pass locally with prod environment variables in the interim.

**Comment by NickAkhmetov** [comment]
TODOs after design review:
- [ ] update timeline to only show 3 items on mobile
- [ ] update "dataset search reorganized" link
- [ ] hide multi-assay datasets item for now
- [ ] fix icon colors for hero
- [ ] add "Data Portal" to end of Hubmap title
- [ ] fix entity counts tooltip so it only shows up when necessary
- [ ] keep aspect ratio of explore tools items on mobile
- [ ] Add icons for explroe tools section

File: context/app/static/storybook-public/mockServiceWorker.js [hunk 1]
```diff
@@ -225,57 +206,46 @@ async function getResponse(event, client, requestId) {
...
     case 'MOCK_RESPONSE': {
       return respondWithMock(clientMessage.data)
     }
 
-    case 'MOCK_NOT_FOUND': {
+    case 'PASSTHROUGH': {
       return passthrough()
     }
-
-    case 'NETWORK_ERROR': {
```
Comment (lchoy) [inline context/app/static/storybook-public/mockServiceWorker.js:265]: Curious why this is removed?

File: context/app/static/js/shared-styles/icons/icons.ts [hunk 2]
```diff
@@ -18,12 +18,15 @@ import ListAltRoundedIcon from '@mui/icons-material/ListAltRounded';
 import ArrowDropDownRoundedIcon from '@mui/icons-material/ExpandMoreRounded';
 import ArrowDropUpRoundedIcon from '@mui/icons-material/ArrowDropUpRounded';
 import AddRoundedIcon from '@mui/icons-material/AddRounded';
-import DescriptionOutlinedIcon from '@mui/icons-material/DescriptionOutlined';
+import DescriptionOutlinedIcon from '@mui/icons-material/DescriptionRounded';
```
Comment (lchoy) [inline context/app/static/js/shared-styles/icons/icons.ts:21]: Could this be confusing if it's renamed?

File: context/app/static/js/components/home/Hero/HeroImageSlide.tsx [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  const theme = useTheme();
+  const isDesktop = useIsDesktop();
+
+  if (!isSlideTitle(title)) {
+    throw new Error(`Invalid title provided to homepage image slide: ${title}`);
+  }
+
+  const { imageBase, imageAlt } = HERO_IMAGE_SLIDES[title];
+  const maxImageWidth = isDesktop ? routeContainerMaxWidth : window.innerWidth - routeContainerPadding;
+  // const maxImageHeight = isDesktop ? 360 : (maxImageWidth * 9) / 32;
```
Comment (lchoy) [inline context/app/static/js/components/home/Hero/HeroImageSlide.tsx:26]: delete?

File: context/app/static/js/components/detailPage/provenance/ProvGraph/ProvGraph.spec.js [hunk 4]
```diff
@@ -17,40 +17,35 @@ document.getElementsByClassName = () => [
...
+        },
       },
-    };
-    return res(ctx.json(descendantData), ctx.status(200));
+    ],
+  },
+};
+
+const server = setupServer(
+  http.post(`/${appProviderEndpoints.elasticsearchEndpoint}`, () => {
```
Comment (lchoy) [inline context/app/static/js/components/detailPage/provenance/ProvGraph/ProvGraph.spec.js:40]: Why `appProviderEndpoints.elasticsearchEndpoint` and not `appProviderEndpoints.searchEndpoint`? The dependency is to search-api.

**Review by lchoy — APPROVED** [review]
Just some small comments and questions

File: context/app/static/storybook-public/mockServiceWorker.js (hunk 1 above)
Comment (NickAkhmetov) [reply context/app/static/storybook-public/mockServiceWorker.js:265]: The `mockServiceWorker.js` file is automatically generated by the `msw` dependency to allow mocking network calls in Storybook/Jest/Cypress: 
image

File: context/app/static/js/components/detailPage/provenance/ProvGraph/ProvGraph.spec.js (hunk 4 above)
Comment (NickAkhmetov) [reply context/app/static/js/components/detailPage/provenance/ProvGraph/ProvGraph.spec.js:40]: This property name was already there (my changes in this file were to update the usage of `mock-service-worker`). We still refer to the search API endpoint as the `elasticsearchEndpoint` in the data that is provided by flask:

image

I've created a ticket to revise this sitewide: https://hms-dbmi.atlassian.net/browse/CAT-710

**Review by lchoy — APPROVED** [review]

**Pull request merged**